JLio.Commands/Advanced/Compare.cs

```diff
@@ -0,0 +1,482 @@
+            if (sourceToken == null || targetToken == null)
+                result.Add(GetStructureDifferenceResult(source, target, propertyName));
+            else
+                result.AddRange(CompareTokens(sourceToken, targetToken));
+            return result;
+        }
+
+        private CompareResult GetStructureDifferenceResult(JObject source, JObject target, string propertyName)
+        {
+            var sourcePath = GetPath(source, options.ItemsFetcher);
+            var targetPath = GetPath(target, options.ItemsFetcher);
+            return new CompareResult
+            {
+                FirstPath = sourcePath,
+                SecondPath = targetPath,
+                DifferenceType = DifferenceType.StructureDifference,
+                DifferenceSubType = eDifferenceSubType.NotEquals,
+                Description =
+                    $"The structure is different. Source: ({sourcePath}{options.ItemsFetcher.PathDelimiter}{propertyName}) -->  {source.ContainsKey(propertyName)} - Target:({targetPath}{options.ItemsFetcher.PathDelimiter}{propertyName}) -->  {target.ContainsKey(propertyName)}",
+                FoundDifference = true
+            };
+        }
+
+        private static IEnumerable<string> GetPropertyNames(JObject source)
+        {
+            return source.Properties().ToList().Select(i => i.Name);
+        }
+
+        private CompareResults CompareArray(JArray source, JArray target)
+        {
+            var result = new CompareResults
+            {
+                source.Count == target.Count
+                    ? GetSameCountOfArrayItemsResult(source, target)
+                    : GetDifferentCountOfArrayItemsResult(source, target)
+            };
+            result.AddRange(CompareArrayItems(source, target));
+            return result;
+        }
+
+        private CompareResults CompareArrayItems(JArray source, JArray target)
+        {
+            var foundTargetItems = new List<int>();
+
+            var result = new CompareResults();
+            foreach (var token in source.Children())
+                result.AddRange(CompareArrayItem(source, target, token, foundTargetItems));
+            for (var i = 0; i < target.Count; i++)
+            {
+                if (foundTargetItems.Contains(i)) continue;
+                result.Add(GetIsInOneOfArraysResult(source, false, target, true, target[i].ToString(Formatting.None)));
+            }
+
+            return result;
+        }
+
+        private CompareResults CompareArrayItem(JArray source, JArray target, JToken sourceItemToCompare,
+            List<int> foundTargetIndexes)
+        {
+            var result = new CompareResults();
+            var settings = Settings.ArraySettings.FirstOrDefault(i => i.ArrayPath == $"$.{source.Path}");
+
+            var IsInTarget = ArrayHelpers.FindInArray(target, sourceItemToCompare, foundTargetIndexes, settings);
+
+            if (IsInTarget.Found) foundTargetIndexes.Add(IsInTarget.Index);
```

Comment:
you have 2 consecutive if statements with the same condition. please combine them into one block